test/Parse/generic_disambiguation.swift

```diff
@@ -76,3 +76,16 @@ A<B?>(x: 0) // parses as type // expected-warning{{unused}}
 _ = a < b ? c : d
 
 A<(B) throws -> D>(x: 0) // expected-warning{{unused}}
+
+// https://github.com/apple/swift/issues/43053
+do {
+  1 as! A<B>.F << 2 // expected-warning{{cast from 'Int' to unrelated type 'A<B>.F' always fails}} // expected-error {{referencing operator function '<<' on 'BinaryInteger' requires that 'A<B>.F' conform to 'BinaryInteger'}}
+  1 as! A<B> << 2 // expected-warning{{cast from 'Int' to unrelated type 'A<B>' always fails}} expected-error{{referencing operator function '<<' on 'BinaryInteger' requires that 'A<B>' conform to 'BinaryInteger'}}
+  let _ = 1 as Int16 << 7
+  let _ = 1 as? Int16 << 7 // expected-error{{value of optional type 'Int16?' must be unwrapped to a value of type 'Int16'}} expected-warning{{conditional downcast from literal to 'Int16' always fails; consider using 'as' coercion}} expected-note{{coalesce using '??' to provide a default when the optional value contains 'nil'}} expected-note{{force-unwrap using '!' to abort execution if the optional value contains 'nil'}}
+  let _ = 1 is Int16 << 7 // expected-warning{{conditional downcast from literal to 'Int16' always fails; consider using 'as' coercion}} expected-error{{binary operator '<<' cannot be applied to operands of type 'Bool' and 'Int'}}
+  let _ = 1 as! Int16 << 7 // expected-warning{{conditional downcast from literal to 'Int16' always fails; consider using 'as' coercion}}
+  let _ = 1 as Int16 <= 7
+  // FIXME: this should not produce any errors.
+  let _ = 1 as Int16 < 7 // expected-error{{expected type}}
+}
```

Comment:
The test suite is not subject to strict formatting, but we tend to place diagnostic verifications on separate lines when a message is long or several verifications target a single line to aid readability (there is an example earlier in this file). I encourage you to follow this unspoken convention.

Reply:
Thank you for pointing out. Will do:)